AsyncDisplayKit/ASMapNode.h

```diff
@@ -0,0 +1,45 @@
+/* Copyright (c) 2014-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+#import <AsyncDisplayKit/AsyncDisplayKit.h>
+#import <MapKit/MapKit.h>
+@interface ASMapNode : ASControlNode
+- (instancetype)initWithCoordinate:(CLLocationCoordinate2D)coordinate NS_DESIGNATED_INITIALIZER;
+/**
+ This is the snapshot shot image node, this will be hidden (but not nil) when .liveMap = YES
+ */
+@property (nonatomic, readonly) ASImageNode *mapImage;
+/**
+ This is the ASDisplayNode that backs the MKMapView. This will be nil if .liveMap = NO. To access the underlying MKMapView, in order to set a delegate for example, use (MKMapView *)mapView.view;
+ */
+@property (nonatomic, readonly) ASDisplayNode *mapView;
+/**
+ Set this to YES to turn the snapshot into an interactive MKMapView and vice versa. Defaults to NO.
+ */
+@property (nonatomic, assign, getter=isLiveMap) BOOL liveMap;
+/**
+ @abstract Explicitly set the size of the map and therefore the size of ASMapNode. Defaults to CGSizeMake(constrainedSize.max.width, 256).
```

Comment:
Why does it default to 256 high?  Is that the right size on an iPad Pro? :)

I think we should default to returning a -layoutSpecThatFits: of an ASRatioLayoutSpec, perhaps.  Maybe there's a better approach? @nguyenhuy  As for the ratio, maybe just a square.  People will have to customize it anyway.

Importantly, the mapView (and if you keep the separate imageNode) will have to be sized to be the full size of the ASMapNode.  The user may choose to set the frame / bounds of the ASMapNode directly to anything, even something that is not the same as what the -layoutSpecThatFits: asks for, and as a framework class we should handle this.

In a way, for this very unusual class, it might not be important to return anything specific for -layoutSpecThatFits: because the map is inherently scalable and has no fixed "form" (unlike text, or even an image with an aspect ratio).  So we might just want to return an empty ASLayoutSpec, and implement a manual -layout method that just sets self.mapView.frame = self.bounds;